### DISCUSSION TITLE
Removing antennae from faces in Arrangement_2

### DISCUSSION THREAD

**Discussion by pentacular** [CGAL/cgal#6726] [Q&A] [answered]
Hello.

I have an arrangement with faces with non-simple boundary polygons.

Looking at the polygon, I see that it is due to an antenna.

```6 vertices: (-0.53 -0.47) (-0.401549 -0.476906) (-0.407594 -0.0356515) (0.186278 -0.00770458) (0.213064 -0.50995) (-0.401549 -0.476906)```

I couldn't see a best practice for cleaning antennae in cgal at the moment.
Have I missed something, or is this something best handled by application code?

**Comment by efifogel** [comment, marked as answer]
Hi Brian,

First, you need to precisely define what an antenna is.
For example, imagine a small polygon strictly inside a large polygon with
an edge connecting the two polygons.
Is the edge an antenna?
If this is not an antenna, then you can simply test whether the face to the
left of an edge e is identical to the face on the right (the left to the
opposite of e), and if so, remove it. Otherwise, it gets a bit more
complicated. You can identify the edges that satisfy the above and the
degree of one endpoint is only 1 (only one incoming and one outgoing
halfedges). For every such edge continue along the antenna through the
other endpoint of the edge.
(Make sure that when you remove edges, you don't do it while traversing the
arrangement. Instead, collect the candidates in a container, and then
traverse and remove the container.)
Anyway, there is no built-in function that does it, but it sounds useful to
me.

Best,
Efi
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by pentacular** [comment]
Ah, right.

I was focusing on the Polygon_2, but you're right that it's much simpler to handle by checking faces in the Arrangement.

Thanks. :)
